.buildkite/scripts/common/run-e2e-tests.sh

```diff
@@ -0,0 +1,27 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+set -x
+
+ROOT=$(dirname "$0")/../../..
+
+# shellcheck disable=SC1091
+source "$ROOT/.env"
+
+# On any command failures, make sure that diagnostics are copied from bucket to local Buildkite environment
+# to be uploaded as artifacts.
+onError() {
+    # On error, copy any artifacts from the test run locally to be picked up as buildkite artifacts
+    if gsutil ls "gs://eck-e2e-buildkite-artifacts/jobs/$CLUSTER_NAME/eck-diagnostic*.zip" 2> /dev/null ; then
+        gsutil cp "gs://eck-e2e-buildkite-artifacts/jobs/$CLUSTER_NAME/eck-diagnostic*.zip" .
+        gsutil rm "gs://eck-e2e-buildkite-artifacts/jobs/$CLUSTER_NAME/eck-diagnostic*.zip" .
+    fi
+}
+
+main() {
+    trap 'onError' ERR
+    make -C .ci get-test-artifacts
+    make e2e-run-actual
```

Comment:
I wonder if we shouldn't instead keep only the gsutil commands in this script and use the bk mechanism `depends_on/allow_failure: true` to trigger it, as we clean up the cluster.

Reply:
Interesting. Hadn't thought of that. Let me see what that would look like and report back.

Reply:
We discussed and decided that having diagnostics files across BK steps wasn't a major issue. I'm working through the changes needed to have the copying of diagnostic files from bucket be triggered by BK cleanup step.

Reply:
update on this: We discussed and decided that having artifacts across multiple BK steps wasn't an issue and adjusted the copying of diagnostics to happen in the cleanup phase of BK.